zerver/webhooks/sonarqube/view.py

```diff
@@ -0,0 +1,104 @@
+from zerver.lib.webhooks.common import check_send_webhook_message
+from zerver.models import UserProfile
+
+inverse_operators = {
+    "WORSE_THAN": "should be better or equal to",
+    "GREATER_THAN": "should be less than or equal to",
+    "LESS_THAN": "should be greater than or equal to",
+}
+
+templates = {
+    "default": "* {}: **{}** {} {} {}.",
+    "no_value": "* {}: **{}**.",
+}
+
+
+def parse_metric_name(metric_name: str) -> str:
+    return " ".join(metric_name.split("_"))
+
+
+def parse_condition(condition: Mapping[str, Any]) -> str:
+    metric = condition["metric"]
+
+    metric_name = parse_metric_name(metric)
+    operator = condition["operator"]
+    operator = inverse_operators.get(operator, operator)
+    value = condition.get("value", "no value")
+    status = condition["status"].lower()
+    threshold = condition["errorThreshold"]
+
+    if value == "no value":
+        return templates["no_value"].format(metric_name, status)
+
+    template = templates["default"]
+
+    return template.format(metric_name, status, value, operator, threshold)
+
+
+def parse_conditions(conditions: List[Mapping[str, Any]]) -> str:
+    return "\n".join(
+        [
+            parse_condition(condition)
+            for condition in conditions
+            if condition["status"].lower() != "ok" and condition["status"].lower() != "no_value"
+        ]
+    )
+
+
+def render_body(payload: Mapping[str, Any]) -> str:
+    project_name = payload["project"]["name"]
+    project_url = payload["project"]["url"]
+    quality_gate_status = payload["qualityGate"]["status"].lower()
+    if quality_gate_status == "ok":
+        quality_gate_status = "success"
+    else:
+        quality_gate_status = "error"
+    branch = None
+    if "branch" in payload.keys():
+        branch = payload["branch"].get("name", None)
+    conditions = payload["qualityGate"]["conditions"]
+    conditions = parse_conditions(conditions)
+    if branch:
+        if not conditions:
+            return "Project [{}]({}) analysis of branch {} resulted in {}.".format(
+                project_name, project_url, branch, quality_gate_status
+            )
+        msg = "Project [{}]({}) analysis of branch {} resulted in {}:\n".format(
+            project_name, project_url, branch, quality_gate_status
+        )
+        msg += conditions
+
+        return msg
+
+    if not conditions:
+        return "Project [{}]({}) analysis resulted in {}.".format(
+            project_name, project_url, quality_gate_status
+        )
+    msg = "Project [{}]({}) analysis resulted in {}:\n".format(
+        project_name, project_url, quality_gate_status
+    )
+    msg += conditions
+
+    return msg
+
+
+@webhook_view("Sonarqube")
+@has_request_variables
+def api_sonarqube_webhook(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    payload: Dict[str, Any] = REQ(argument_type="body"),
+) -> HttpResponse:
+    topic = payload["project"]["name"]
```

Comment:
Another potential improvement here is to include the branch name in the topic. That would lead to messages for the same branch being grouped under the same topic and simplifying a lot of the logic in `render_body` as well. Ideally, you want to easily be able to see the analysis notifications for `master` under the same topic. This can be a part of a follow-up PR as well!

Reply:
@timabbott Not sure if you have any thoughts on this. In general, grouping messages by branch seems to be something we do a lot in our integrations. I think it is a good principle to adhere to consistently across integrations!

Reply:
I added the branch to topic similar to what we do for github integration. Here is the screenshot
![001](https://user-images.githubusercontent.com/59223034/115963553-6de14500-a53d-11eb-8548-e581feaab7aa.png)